aten/src/ATen/gen_backend_select_register.py

```diff
@@ -0,0 +1,64 @@
+# This script generates BackendSelectRegister.cpp which is being used for dispatching purposes.
+# We process only those factory functions that have 'backend_select' flag in its native_functions.yaml definition.
+
+from code_template import CodeTemplate
+
+GENERATED_COMMENT = CodeTemplate(
+    "@" + "generated from ${filename}")
+
+FUNCTION_REGISTRATION = CodeTemplate("""\
+.op(torch::RegisterOperators::options()
+  .schema("${schema_string}")
+  .impl_unboxedOnlyKernel<decltype(${function_name}), &${function_name}>(DispatchKey::BackendSelect)
+  .aliasAnalysis(AliasAnalysisKind::FROM_SCHEMA))
+""")
+
+FUNCTION_DEFINITION = CodeTemplate("""\
+Tensor ${function_name}(${method_formals}) {
+  DispatchKey key = options.computeDispatchKey();
+  static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::${name}", "${overload_name}");
+  return op.callUnboxedWithDispatchKey<${formals_types}>(key, ${type_method_actuals});
+}
+""")
+
+def needs_backend_select(declaration_option):
+    # We register an op under the BackendSelect dispatch key
+    # if a TensorOptions argument has been gathered from its declared args
+    # We skip all the 'new_*' and '*_like' ops as they are special cased and avoid dispatching.
+    # See TypeDefault.cpp
+    if '_like' in declaration_option['name'] or 'new_' in declaration_option['name']:
+        return False
+
+    return any("dynamic_type" in a and a["dynamic_type"] == "TensorOptions" for a in declaration_option["arguments"])
```

Comment:
it's more idiomatic to do something like:
`return any(a.get('dynamic_type') == 'TensorOptions' for a in declaration_option['arguments'])`

Reply:
Updated